gluten-core/src/main/scala/io/glutenproject/execution/LimitTransformer.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.execution
+
+import com.google.common.collect.Lists
+import com.google.protobuf.Any
+import io.glutenproject.GlutenConfig
+import io.glutenproject.expression.ConverterUtils
+import io.glutenproject.substrait.SubstraitContext
+import io.glutenproject.substrait.`type`.{TypeBuilder, TypeNode}
+import io.glutenproject.substrait.extensions.ExtensionBuilder
+import io.glutenproject.substrait.plan.PlanBuilder
+import io.glutenproject.substrait.rel.{RelBuilder, RelNode}
+import io.glutenproject.utils.BindReferencesUtil
+import io.glutenproject.vectorized.ExpressionEvaluator
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.execution.{SparkPlan, UnaryExecNode}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+import java.util
+
+case class LimitTransformer(
+                             child: SparkPlan,
+                             offset: Long,
+                             count: Long)
+  extends UnaryExecNode
```

Comment:
It seems extending LimitExec is better, right?

Reply:
We need push the limit operation down into native engine in the future.